### PR TITLE
8254599: CDS dump should not warn about hidden classes

### PR DESCRIPTION
**Pull request by yminqi**
Regular hidden class skipping warning in -Xshare:dump is at info log level, it is noise for a normal operation. Changed to debug level, along the change, three test cases in dynamic dumping changed to debug log level to have original message output. 

Tests: tier1-5 in progress. Passed all local runtime/cds

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8254599](https://bugs.openjdk.java.net/browse/JDK-8254599): CDS dump should not warn about hidden classes

### Reviewers
 * [Claes Redestad](https://openjdk.java.net/census#redestad) (@cl4es - **Reviewer**) ⚠️ Review applies to ded59cf26765673238a1d7650a89825400569adf
 * [Lois Foltan](https://openjdk.java.net/census#lfoltan) (@lfoltan - **Reviewer**) ⚠️ Review applies to ded59cf26765673238a1d7650a89825400569adf
 * [Ioi Lam](https://openjdk.java.net/census#iklam) (@iklam - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/614/head:pull/614`
`$ git checkout pull/614`

### CODE REVIEW

**Review by cl4es — APPROVED** [review]

**Review by lfoltan — APPROVED** [review]
Looks good Yumin!

**Comment by yminqi** [comment]
Thanks Claes and Lois for the review!

File: src/hotspot/share/classfile/systemDictionaryShared.cpp [hunk 1]
```diff
@@ -1346,7 +1346,7 @@ bool SystemDictionaryShared::should_be_excluded(InstanceKlass* k) {
   }
 
   if (k->is_hidden() && !is_registered_lambda_proxy_class(k)) {
-    warn_excluded(k, "Hidden class");
+    log_debug(cds)("Skipping %s: %s", k->name()->as_C_string(), "Hidden class");
```
Comment (iklam) [inline src/hotspot/share/classfile/systemDictionaryShared.cpp:1349]: A ResourceMark is needed here for calling k->name()->as_C_string().

**Review by iklam — CHANGES_REQUESTED** [review]

File: src/hotspot/share/classfile/systemDictionaryShared.cpp (hunk 1 above)
Comment (yminqi) [reply src/hotspot/share/classfile/systemDictionaryShared.cpp:1350]: Thanks, yes here needs a ResourceMark.

**Review by iklam — APPROVED** [review]
LGTM.

**Pull request closed without merging**